src/components/GeneralInfo/DataCollectorsCard/DataCollectorsCard.js

```diff
@@ -0,0 +1,123 @@
+                    <Tr>
+                        {collector.details.name ?
+                            <Td
+                                expand={
+                                    collector.details
+                                        ? {
+                                            rowIndex,
+                                            isExpanded: isExpanded(collector),
+                                            onToggle: () => setExpanded(collector, !isExpanded(collector))
+                                        }
+                                        : undefined
+                                }
+                                style={{ paddingLeft: 0 }}
+                            /> : <Td />}
+                        <Td dataLabel={'Name'}>{collector.name}</Td>
+                        <Td dataLabel={'Status'}>{collector.status ? 'Active' : 'N/A'}</Td>
+                        <Td dataLabel={'Last upload'}>
+                            {!collector.updated && 'N/A'}
+                            {DateFormat && collector.updated ?
+                                <DateFormat date={ collector.updated } type="exact" /> :
+                                (collector.updated && new Date(collector.updated).toLocaleString())
+                            }</Td>
+                    </Tr>
+                    {collector.details && collector.details.name && (
+                        <Tr isExpanded={isExpanded(collector)}>
+                            <Td />
+                            <Td colSpan={3}>
+                                <ExpandableRowContent
+                                    style={{ paddingTop: 10, paddingBottom: 15, paddingLeft: 15 }}>
+                                    <Flex>
+                                        <FlexItem style={{ marginRight: 5 }}>
+                                            {`${collector.details.name}:`}
+                                        </FlexItem>
+                                        <FlexItem grow={{ default: 'grow' }}>
+                                            {collector.details.id ?? 'N/A'}
+                                        </FlexItem>
+                                    </Flex>
+                                </ExpandableRowContent>
+                            </Td>
+                        </Tr>
+                    )}
+                </Tbody>))}
+        </TableComposable>
+    </LoadingCard>);
+};
+
+DataCollectorsCard.propTypes = {
+    detailLoaded: PropTypes.bool,
+    collectors: PropTypes.array,
+    entity: PropTypes.shape({
+        per_reporter_staleness: PropTypes.object
+    })
+};
+DataCollectorsCard.defaultProps = {
+    detailLoaded: false
+};
+
+export default connect(({
+    entityDetails: {
+        entity
+    },
+    systemProfileStore: {
+        systemProfile
+    }
+}) => ({
+    entity,
+    systemProfile,
+    detailLoaded: systemProfile && systemProfile.loaded
```

Comment:
Safe accessor can be used here
```suggestion
    detailLoaded: systemProfile?.loaded
```